.github/workflows/ci.yml

```diff
@@ -73,8 +74,15 @@ jobs:
         run: git config --system core.longpaths true
       - id: checkout
         uses: actions/checkout@v3
-      - name: Set up MSBuild
-        uses: microsoft/setup-msbuild@v1.1
+      - name: Run sccache-cache
+        uses: mozilla-actions/sccache-action@v0.0.3
+      - name: sccache cache
+        uses: actions/cache@v3
+        with:
+          path: ~/AppData/Local/Mozilla/sccache/cache
+          key: ${{ runner.os }}-sccache
```

Comment:
Done, but there is a problem with this approach: It requires multiple times as much storage as a single central cache: one for main, and one for each active PR. We're already hitting the 10 GB limit, caches are getting evicted, and we're on the way to cache trashing, but not quite there yet.

We will need a custom action in the future, that drops older caches in a smart way, before GitHub drops the oldest, without considering the branch. https://docs.github.com/en/actions/using-workflows/caching-dependencies-to-speed-up-workflows#force-deleting-cache-entries

Added the ref, but separating caches by SHA seems excessive: that's a separate cache key for each commit, raising the number of caches in storage from the number of active PRs * number of platforms to the number of commits * number of platforms.

Reply:
https://github.com/apache/nifi-minifi-cpp/actions/caches

Reply:
Unfortunately that will not work due to the [caches being immutable](https://github.com/actions/cache/blob/04f198bf0b2a39f7230a4304bf07747a0bddf146/tips-and-workarounds.md?plain=1#L11). This is why it is stated in the documentation to use unique keys for cache entries (using the SHA for example). Cache entries cannot be overwritten, so if a key is set to save to an already existing cache entry key it is dropped and the cache for that key will not be updated, it will produce the following message:

```
Cache hit occurred on the primary key ubuntu-20.04-ccache-refs/heads/MINIFICPP-2127_ci, not saving cache.
```

As you said it would be better to have a custom action to clean up old unused caches, but not only for dropped branches but for the obsolete cache entries in the same branch as well that has an updated cache entry present already. For now it is better to have a unique identifier for the cache keys to have more up to date entries, but we may figure out a way in the future to have an automatic cache cleanup procedure for obsolete entries.